### PR TITLE
Feature/horizontal waves animation

### PR DESCRIPTION
**Pull request by jonasgrootkormelink**

### CODE REVIEW

File: client/src/nl/sensorlab/videowall/animation/baseanimations/horizontalwaves/HorizontalWave.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+		// Reset intensity to keep within bounds
+		offsetIntensity = 0;
+		
+		g.noFill();
+		g.stroke(color);
+		g.strokeWeight(1);
+		
+		// Start the shape
+		g.beginShape();
+		for(int x = 0; x <= parent.PIXEL_RESOLUTION_X; x++) {
```
Comment (wilcotomassen) [inline client/src/nl/sensorlab/videowall/animation/baseanimations/horizontalwaves/HorizontalWave.java]: Use a static accessor here: e.g. BaseAnimation.PIXEL_RESOLUTION_X

File: client/src/nl/sensorlab/videowall/animation/baseanimations/horizontalwaves/HorizontalWave.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+			// Returns the Perlin noise value at specified coordinates. 
+			// Perlin noise is a random sequence generator producing a more natural, harmonic succession of numbers than that of the standard random() function. 
+			// It was developed by Ken Perlin in the 1980s and has been used in graphical applications to generate procedural textures, shapes, terrains, and other seemingly organic forms.
+			float perlinNoise = parent.parent.noise(offsetIntensity, offsetSpeed);
+			float y = parent.parent.map(perlinNoise, 0, 1, 0, parent.PIXEL_RESOLUTION_Y);
```
Comment (wilcotomassen) [inline client/src/nl/sensorlab/videowall/animation/baseanimations/horizontalwaves/HorizontalWave.java]: Use a static accessor here as well

File: client/src/nl/sensorlab/videowall/animation/baseanimations/horizontalwaves/HorizontalWave.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+			g.vertex(x, y);
+			
+			// Update the offset intensity
+			offsetIntensity += offsetIntensityIncrements;
+		}
+		// Close the shape
+		g.endShape();
+		
+		// Increase the offset so it creates a wave
+		offsetSpeed += offsetSpeedIncrements;
```
Comment (wilcotomassen) [inline client/src/nl/sensorlab/videowall/animation/baseanimations/horizontalwaves/HorizontalWave.java]: Use delta time here to decouple animation speed and framerate

**Review by wilcotomassen — CHANGES_REQUESTED** [review]
First: fix the merge conflicts with develop (probably just the gitignore and animationManager

Then: Can you move HorizontalWave into HorizontalWavesAnimation and remove the horizontalwaves package? Makes it all a bit more compact

**Comment by jonasgrootkormelink** [comment]
@wilcotomassen I think we can merged this branch into develop. I've made the requested changes!

**Review by wilcotomassen — APPROVED** [review]

**Pull request merged**